### PR TITLE
Last Call updates to draft-ietf-cose-key-thumbprint.md

### LINKED ISSUE #95: AD review draft-ietf-cose-key-thumbprint-04
In the Security Considerations:

> To promote interoperability among implementations, the SHA-256 hash algorithm is mandatory to implement.

- [ ] This really belongs somewhere in the main specification document, and not in the Security Consideration. Someone should be able to implement the spec without reading the Security Considerations.

> Using thumbprints with passwords (i.e. low-entropy secrets) is dangerous and MUST be avoided.

- [ ] "MUST be avoided" is an odd expression and leaves some wiggle room. ("it was unavoidable, so I did it anyway"). Can it not more plainly say "Thumbprints MUST NOT be used with passwords" ?

### PR DESCRIPTION
**Pull request by OR13**
Closes #95 

Will still need to run ID Nits and address line breaks comment before publishing new version.

### CODE REVIEW

**Review by selfissued — APPROVED** [review]

**Comment by hannestschofenig** [comment]
Looks good to me. Thanks for making the changes

**Review by ko-isobe — APPROVED** [review]

**Comment by ko-isobe** [comment]
I agree with this improvement.

File: cose-key-thumbprint/draft-ietf-cose-key-thumbprint.md [hunk 1]
```diff
@@ -464,19 +464,19 @@ IANA is requested to add the following entry to the IANA "CWT Confirmation
 Methods" registry established by {{RFC8747}}:
 
 - Confirmation Method Name: ckt
-- Confirmation Method Description: COSE Key Thumbprint
+- Confirmation Method Description: COSE Key SHA-256 Thumbprint
 - JWT Confirmation Method Name: jkt
```
Comment (selfissued) [inline cose-key-thumbprint/draft-ietf-cose-key-thumbprint.md:468]:
```suggestion
- Confirmation Method Name: jkt

```

Comment (OR13) [reply cose-key-thumbprint/draft-ietf-cose-key-thumbprint.md:468]: Reading from the registration template in: https://datatracker.ietf.org/doc/html/rfc8747#section-7.2.1

```
JWT Confirmation Method Name:
Claim Name of the equivalent JWT confirmation method value, as registered in the "JSON Web Token Claims" subregistry in the "JSON Web Token (JWT)" registry [[IANA.JWT](https://www.iana.org/assignments/jwt)]. CWT claims should normally have a corresponding JWT claim. If a corresponding JWT claim would not make sense, the designated experts can choose to accept registrations for which the JWT Claim Name is listed as "N/A".
```

I think this is correct as is

Comment (OR13) [reply cose-key-thumbprint/draft-ietf-cose-key-thumbprint.md:468]: The intention is to keep JWT and CWT confirmation methods aligned, so CWT registrations are requested to cite JWT ones.

**Comment by OR13** [comment]
We should merge this an publish a new version

**Comment by OR13** [comment]
I'm merging this, so I can address https://datatracker.ietf.org/doc/review-ietf-cose-key-thumbprint-04-genart-lc-knodel-2024-04-01/ in a subsequent PR

**Pull request merged**